docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -374,43 +373,114 @@ In our architecture, each scheduler has global view of resources and runs its sc
+The Scheduling Event Handler will notify the VM/Container Status Watcher when it retrieves a VM/Container scheduling request. The VM/Container Status Watcher will start a timer for this VM/Container scheduling. If within this timer period, it does not receive a “success” status notification from the selected cluster, it will treat this as a scheduling failure. 
+When the VM/Container Status Watcher receives a “scheduling failure”, either from the cluster’s Status Agent or from the timeout routine, it will send a “delete” message to the VM/Container’s previously selected cluster by queueing a “delete” event to the Scheduling Work Queue. Then it will put the VM/Container back to the head of the Scheduling Work Queue for rescheduling. To prevent a second-time failure, the previously selected cluster will be assigned a very low score in the scheduler’s ranking phase or even be removed from this VM/Container’s candidate list if the candidate list has other clusters
+
+### 3.3.2 Pull Model
+This model does not require a VM/Container Status Agent to be placed into each cluster. The VM/Container Status Watcher will start a query timer and query/pull each cluster for the VM’s scheduling status information every timer interval. 
+The interface and transport protocol between the VM/Container Status Watcher and the cluster depends on the type of API each cluster supports. 
+This model will introduce a lot of unnecessary traffic into the control plane’s network if the pulling interval is too small. If the pulling interval is too large, the end-to-end scheduling latency will be high. It is a dilemma to choose a proper pulling interval. The VM/Container Status Watcher actually runs in a kind of “blindly” pull mode.   
+
+## 3.4 Cluster Resource Collection Flow
+This refers to the flow from Cluster->Cluster Resource Collector->API Server->Dynamic Resource Infor Cache DB. Similar to what is described in section 3.3, there are two ways to design and implement the resource collection flow as described in the following two sections. 
+
+### 3.4.1 Push Model (Preferred Model)
+This model requires the cluster to proactively send its resource information to the global scheduling framework’s Cluster Resource Collector. The cluster will only send its resource information when there is a resource change. If network traffic is a concern, “batch operation” can be used, i.e. resource update will be accumulated for a pre-defined interval and then sent to the global scheduling framework. This prevents injecting a burst of “update” traffic into the control plane’s network when the resource change frequency is very high at some time period. This interval value should be defined to keep a good balance between “accurate in-sync of global scheduling framework’s Dynamic Resource Info Cache with each cluster’s actual resource info DB” and “low network traffic and database/cache access traffic”
```

Comment:
The traffic flow need to be considered  independently,it's another topic which is very different from resource scheduler.

Reply:
As discussed in the open source meeting, we will go for the push mode flow. In this mode, the cluster resource agent has knowledge about when and how often the resource changes in each cluster and can better decide when to send an update of resource to the global scheduling framework. We will use this push mode flow.